### PR TITLE
contrib/aspell: new package (0.60.8.1) 

### PR DESCRIPTION
**Pull request by ifreund**

### CODE REVIEW

File: contrib/aspell-en/template.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+make_cmd = "gmake"
+hostmakedepends = [
+    "aspell",
+    "gmake",
+]
+pkgdesc = "English dictionary for aspell"
+maintainer = "Isaac Freund <mail@isaacfreund.com>"
+license = "custom:aspell-en"
+url = "http://aspell.net"
+source = f"https://ftp.gnu.org/gnu/aspell/dict/en/aspell6-en-2020.12.07-0.tar.bz2"
```
Comment (nekopsykose) [inline contrib/aspell-en/template.py:14]:
```suggestion
source = f"https://ftp.gnu.org/gnu/aspell/dict/en/aspell6-en-{pkgver}-0.tar.bz2"
```

**Comment by q66** [comment]
is there a real purpose to this? i kinda intentionally did not add it because it seemed pointless

**Comment by ifreund** [comment]
> is there a real purpose to this? i kinda intentionally did not add it because it seemed pointless

It's the only spell checker supported by kakoune's bulitin spell checking: https://github.com/mawww/kakoune/blob/e34735a35041c19cc80d24cab6237abd447b8924/rc/tools/spell.kak

**Comment by NovaAndrom3da** [comment]
@nekopsykose weren't you at one point replacing any instances of "{pkgname}" in sources with the actual package name?

**Comment by NovaAndrom3da** [comment]
contrib/aspell-en needs an update.py file with:

```
pkgname="aspell"
```

File: contrib/aspell/template.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "automake",
+    "gettext-devel",
+    "gmake",
+    "libtool",
+]
+pkgdesc = "Spell checker with good multi-language support"
+maintainer = "Isaac Freund <mail@isaacfreund.com>"
+license = "LGPL-2.1-only"
+url = "http://aspell.net"
+source = f"https://ftp.gnu.org/gnu/{pkgname}/{pkgname}-{pkgver}.tar.gz"
```
[the remark this answers is not in the document]
Comment (NovaAndrom3da) [reply contrib/aspell/template.py:16]: oops, looks like i had forgotten to hit submit on my review

**Comment by q66** [comment]
move to main and add support to enchant

File: contrib/aspell/template.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ifreund) [reply contrib/aspell/template.py]: sure

**Comment by ifreund** [comment]
> move to main and add support to enchant

Done :)

File: main/aspell/template.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+]
+pkgdesc = "Spell checker with good multi-language support"
+maintainer = "Isaac Freund <mail@isaacfreund.com>"
+license = "LGPL-2.1-only"
+url = "http://aspell.net"
+source = f"https://ftp.gnu.org/gnu/aspell/aspell-{pkgver}.tar.gz"
+sha256 = "d6da12b34d42d457fa604e435ad484a74b2effcd120ff40acd6bb3fb2887d21b"
+
+
+@subpackage("libaspell")
```
Comment (q66) [inline main/aspell/template.py:20]: should be called aspell-libs

Comment (ifreund) [reply main/aspell/template.py:20]: There's only one shared library in the package, `libaspell.so`. According to Packaging.md `libaspell` is correct: 

> However, if a library is a subpackage of a bigger software project, there are two things you can do. If the subpackage provides a single library that is usable as a standalone runtime dependency for other things, you should use the `lib` prefix.

I don't care strongly one way or the other, the docs should probably be updated if they're out of date though.

Comment (ifreund) [reply main/aspell/template.py]: ah indeed, I think my eyes glazed over the single letter difference or something. Done

**Pull request merged**